src/styles/index.scss

```diff
@@ -1,3 +1,78 @@
 body {
   margin: 0;
+  min-height: 100vh;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  box-sizing: border-box;
+}
+
+.calendar {
+  padding: 10px;
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  width: 706px;
+
+  // .calendar__item
+
+  &__day {
+    box-sizing: border-box;
+    list-style: none;
+    height: 100px;
+    width: 100px;
+    background-color: #eee;
+    border: 1px solid black;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+
+    &:hover {
+      background-color: #ffbfcb;
+      transform: translateY(-20px);
+      animation: move 0.5s;
+    }
+  }
+
+  @for $number from 1 through 31 {
+    &:nth-child(#{$number})::before {
+      content: '#{$number}';
+      font-family: Arial, Helvetica, sans-serif;
+      font-size: 30px;
+    }
+  }
+
+  $days: (
+    'mon' 0,
+    'tue' 1,
+    'wed' 2,
+    'thu' 3,
+    'fri' 4,
+    'sat' 5,
+    'sun' 6
+  );
+
+@each $day, $dayCount in $days {
+  &--start-day-#{$day}:first-child {
+    margin-left: 101px * $dayCount;
+  }
+}
+
+@for $number from 28 through 31 {
+  &--month-length-#{$number}:nth-child(n + #{$number + 1}) {
```

Comment:
Remove the space between `&--month-length-#{$number}` and `:nth-child(n + #{$number + 1})` to avoid specificity issues. This will ensure that your styles are applied correctly.